core/trino-main/src/main/java/io/trino/sql/gen/JoinCompiler.java

```diff
@@ -1019,12 +1021,14 @@ private BytecodeNode typeEqualsIgnoreNulls(
     {
         private final Constructor<? extends LookupSourceSupplier> constructor;
         private final PagesHashStrategyFactory pagesHashStrategyFactory;
+        private final boolean singleBigintColumn;
```

Comment:
remove. just make local decision in `createLookupSourceSupplier` below

Reply:
There are no types available in `createLookupSourceSupplier` so no way of changing whether it is bigint

Reply:
Can you pass `OptionalInt singleBigintJoinChannel` then here?